plugins/woo-ai/includes/class-woo-ai-settings.php

```diff
@@ -0,0 +1,217 @@
+	 */
+	protected static $instance = null;
+
+
+	/**
+	 * Settings ID.
+	 *
+	 * @var id
+	 */
+	protected $id = 'woo-ai-settings-tab';
+
+	/**
+	 * Main Instance.
+	 */
+	public static function instance() {
+		self::$instance = is_null( self::$instance ) ? new self() : self::$instance;
+		return self::$instance;
+	}
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		add_action( 'admin_enqueue_scripts', array( $this, 'add_woo_ai_settings_script' ) );
+		add_action( 'woocommerce_settings_save_advanced', array( $this, 'action_save_woo_ai_settings_tab' ) );
+		add_action( 'woocommerce_settings_page_init', array( $this, 'add_ui' ) );
+
+		$this->add_sanitization_hooks();
+	}
+
+	/**
+	 * Add UI related hooks.
+	 */
+	public function add_ui() {
+		add_filter( 'woocommerce_get_settings_advanced', array( $this, 'add_woo_ai_settings' ), 10, 2 );
+	}
+
+	/**
+	 * Save settings.
+	 */
+	public function action_save_woo_ai_settings_tab() {
+		WC_Admin_Settings::save_fields(
+			$this->get_woo_ai_settings()
+		);
+	}
+
+	/**
+	 * Add sanitization hooks.
+	 */
+	public function add_sanitization_hooks() {
+		$settings = $this->get_woo_ai_settings();
+
+		foreach ( $settings as $setting ) {
+			if ( in_array( $setting['type'], array( 'text', 'textarea' ), true ) ) {
+				add_filter( 'woocommerce_admin_settings_sanitize_option_' . $setting['id'], array( $this, 'strip_tags_field_value' ) );
+			}
+		}
+	}
+
+	/**
+	 * Sanitize field value.
+	 *
+	 * @param string $raw_value The current section.
+	 */
+	public function strip_tags_field_value( $raw_value ) {
+		return wp_strip_all_tags( $raw_value ?? '' );
+	}
+
+	/**
+	 * Add settings to the AI section.
+	 *
+	 * @param array  $settings The original settings array.
+	 * @param string $current_section The current section.
+	 */
+	public function add_woo_ai_settings( $settings = array(), $current_section = null ) {
+		if ( 'features' !== $current_section ) {
+			return $settings;
+		}
+
+		return array_merge(
+			$this->get_woo_ai_settings(),
+			$settings
+		);
+	}
+
+	/**
+	 * Return array describing new settings.
+	 */
+	public function get_woo_ai_settings() {
+
+		// These are made available in the DOM and actually toggled client-side.
+		$tone_desc = array(
+			'informal'     => __( 'Relaxed and friendly, as if you were having a conversation with a friend.', 'woocommerce' ),
+			'humorous'     => __( 'Engage customers with a light-hearted and fun communication style.', 'woocommerce' ),
+			'neutral'      => __( 'A balanced and impartial tone that uses casual expressions without slang.', 'woocommerce' ),
+			'youthful'     => __( 'Bring energy and enthusiasm to your store with a friendly and cheeky tone.', 'woocommerce' ),
+			'formal'       => __( 'Direct yet respectful, a formal tone sounds serious and professional.', 'woocommerce' ),
+			'motivational' => __( 'Passionate and inspiring, engage users with this upbeat tone.', 'woocommerce' ),
+		);
+
+		$markup_str = array_reduce(
```

Comment:
I wonder if we could use the WC Core Settings API which has some utilities for adding Select fields, it might not require us to manually build the HTML markup. 🤔 
https://woocommerce.com/document/settings-api/
Some examples from the WC org:
https://github.com/search?q=org%3Awoocommerce%20woocommerce_admin_fields&type=code

Reply:
@tommyshellberg So, the markup here is just to enable the client-side behavior of showing the option descriptions when they're selected. They're hidden on page load, and this the client-side logic just displays the one that matches the option that is selected. 

I'm  not sure I see a utility here that would assist that, but I could totally be missing something! 🤔

Reply:
Ah, okay, I misread this, then. The settings themselves are added through the `woocommerce_get_settings_advanced` filter and the markup string is added to the select field description.
Clever!
Carry on 😄